client/src/util/ensureFlixExists.ts

```diff
@@ -9,23 +9,16 @@ import { USER_MESSAGE } from './userMessages'
 const FLIX_JAR = 'flix.jar'
 
 async function downloadWithRetryDialog<T>(downloadFunc: () => Promise<T>): Promise<T> {
-  while (true) {
-    try {
-      return await downloadFunc()
-    } catch (e) {
-      const { msg, option1, option2 } = USER_MESSAGE.ASK_DOWNLOAD_RETRY(e.message)
-      const selected = await vscode.window.showErrorMessage(
-        msg,
-        {},
-        { title: option1, retry: true },
-        { title: option2 },
-      )
+  try {
+    return await downloadFunc()
+  } catch (e) {
+    const { msg, option1, option2 } = USER_MESSAGE.ASK_DOWNLOAD_RETRY(e.message)
+    const selected = await vscode.window.showErrorMessage(msg, {}, { title: option1, retry: true }, { title: option2 })
 
-      if (selected?.retry) {
-        continue
-      }
-      throw e
+    if (selected?.retry) {
+      downloadWithRetryDialog(downloadFunc)
```

Comment:
There is no return here... so you always throw.

Reply:
Also not sure why recursion is more clear than a loop?

Reply:
Ah - Missed that. This was because of a lint disallowing constants in conditionals, hence no `while (true)`. This was a way to get around having a `done` variable or something like that, assuming that the number of retries isn't gonna be super high. 
An alternative is to simply disable this lint.

Reply:
I suggest to keep some form of loop variable or disable the lint.